core/src/main/kotlin/net/corda/core/serialization/internal/AttachmentsClassLoader.kt

```
@@ -470,7 +470,13 @@ interface AttachmentsClassLoaderCache {
 @DeleteForDJVM
 class AttachmentsClassLoaderCacheImpl(cacheFactory: NamedCacheFactory) : SingletonSerializeAsToken(), AttachmentsClassLoaderCache {
 
-    private val cache: Cache<AttachmentsClassLoaderKey, SerializationContext> = cacheFactory.buildNamed(Caffeine.newBuilder(), "AttachmentsClassLoader_cache")
+    private val cache: Cache<AttachmentsClassLoaderKey, SerializationContext> = cacheFactory.buildNamed(
+        // Close deserialization classloaders when we evict them
+        // to release any resources they may be holding.
+        Caffeine.newBuilder().removalListener { _, context, _ ->
+            (context?.deserializationClassLoader as? AutoCloseable)?.use {}
```

I think we should call `close` instead of `use`. We don't need the lambda running feature of `use` and `close` makes our intention clearer.

---

The point was that `close()` can itself throw `Exception`, which we would also need to handle. Kotlin's `use {}` function invokes `closeFinally()` to "suppress possible exceptions thrown by AutoCloseable.close()". However, looking more closely at `closeFinally()`'s implementation shows that it doesn't do that I wanted anyway.

I'll replace the code with this:
```kotlin
try {
    (context?.deserializationClassLoader as? AutoCloseable)?.close()
} catch (_: Exception) {
}
```